// typeorm-extension/dist/env/constants.js
"use strict";
/*
 * Copyright (c) 2023-2023.
 * Author Peter Placzek (tada5hi)
 * For the full copyright and license information,
 * view the LICENSE file that was distributed with this source code.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvironmentVariableName = exports.EnvironmentName = void 0;
var EnvironmentName;
(function (EnvironmentName) {
    EnvironmentName["DEVELOPMENT"] = "development";
    EnvironmentName["PRODUCTION"] = "production";
    EnvironmentName["TEST"] = "test";
})(EnvironmentName = exports.EnvironmentName || (exports.EnvironmentName = {}));
var EnvironmentVariableName;
(function (EnvironmentVariableName) {
    EnvironmentVariableName["ENV"] = "NODE_ENV";
    // Seeder
    EnvironmentVariableName["SEEDS"] = "DB_SEEDS";
    EnvironmentVariableName["SEEDS_ALT"] = "TYPEORM_SEEDING_SEEDS";
    EnvironmentVariableName["FACTORIES"] = "DB_FACTORIES";
    EnvironmentVariableName["FACTORIES_ALT"] = "TYPEORM_SEEDING_FACTORIES";
    // Database
    EnvironmentVariableName["TYPE"] = "DB_TYPE";
    EnvironmentVariableName["TYPE_ALT"] = "TYPEORM_CONNECTION";
    EnvironmentVariableName["URL"] = "DB_URL";
    EnvironmentVariableName["URL_ALT"] = "TYPEORM_URL";
    EnvironmentVariableName["HOST"] = "DB_HOST";
    EnvironmentVariableName["HOST_ALT"] = "TYPEORM_HOST";
    EnvironmentVariableName["PORT"] = "DB_PORT";
    EnvironmentVariableName["PORT_ALT"] = "TYPEORM_PORT";
    EnvironmentVariableName["USERNAME"] = "DB_USERNAME";
    EnvironmentVariableName["USERNAME_ALT"] = "TYPEORM_USERNAME";
    EnvironmentVariableName["PASSWORD"] = "DB_PASSWORD";
    EnvironmentVariableName["PASSWORD_ALT"] = "TYPEORM_PASSWORD";
    EnvironmentVariableName["DATABASE"] = "DB_DATABASE";
    EnvironmentVariableName["DATABASE_ALT"] = "TYPEORM_DATABASE";
    EnvironmentVariableName["SID"] = "DB_SID";
    EnvironmentVariableName["SID_ALT"] = "TYPEORM_SID";
    EnvironmentVariableName["SCHEMA"] = "DB_SCHEMA";
    EnvironmentVariableName["SCHEMA_ALT"] = "TYPEORM_SCHEMA";
    EnvironmentVariableName["SCHEMA_DROP"] = "DB_DROP_SCHEMA";
    EnvironmentVariableName["SCHEMA_DROP_ALT"] = "TYPEORM_DROP_SCHEMA";
    EnvironmentVariableName["DRIVER_EXTRA"] = "DB_DRIVER_EXTRA";
    EnvironmentVariableName["DRIVER_EXTRA_ALT"] = "TYPEORM_DRIVER_EXTRA";
    EnvironmentVariableName["SYNCHRONIZE"] = "TYPEORM_SYNCHRONIZE";
    EnvironmentVariableName["SYNCHRONIZE_ALT"] = "TYPEORM_SYNCHRONIZE";
    EnvironmentVariableName["MIGRATIONS"] = "DB_MIGRATIONS";
    EnvironmentVariableName["MIGRATIONS_ALT"] = "TYPEORM_MIGRATIONS";
    EnvironmentVariableName["MIGRATIONS_RUN"] = "DB_MIGRATIONS_RUN";
    EnvironmentVariableName["MIGRATIONS_RUN_ALT"] = "TYPEORM_MIGRATIONS_RUN";
    EnvironmentVariableName["MIGRATIONS_TABLE_NAME"] = "DB_MIGRATIONS_TABLE_NAME";
    EnvironmentVariableName["MIGRATIONS_TABLE_NAME_ALT"] = "TYPEORM_MIGRATIONS_TABLE_NAME";
    EnvironmentVariableName["ENTITIES"] = "TYPEORM_ENTITIES";
    EnvironmentVariableName["ENTITIES_ALT"] = "TYPEORM_ENTITIES";
    EnvironmentVariableName["ENTITY_PREFIX"] = "DB_ENTITY_PREFIX";
    EnvironmentVariableName["ENTITY_PREFIX_ALT"] = "TYPEORM_ENTITY_PREFIX";
    EnvironmentVariableName["METADATA_TABLE_NAME"] = "DB_METADATA_TABLE_NAME";
    EnvironmentVariableName["METADATA_TABLE_NAME_ALT"] = "TYPEORM_METADATA_TABLE_NAME";
    EnvironmentVariableName["SUBSCRIBERS"] = "DB_SUBSCRIBERS";
    EnvironmentVariableName["SUBSCRIBERS_ALT"] = "TYPEORM_SUBSCRIBERS";
    EnvironmentVariableName["LOGGING"] = "DB_LOGGING";
    EnvironmentVariableName["LOGGING_ALT"] = "TYPEORM_LOGGING";
    EnvironmentVariableName["LOGGER"] = "DB_LOGGER";
    EnvironmentVariableName["LOGGER_ALT"] = "TYPEORM_LOGGER";
    EnvironmentVariableName["MAX_QUERY_EXECUTION_TIME"] = "DB_MAX_QUERY_EXECUTION_TIME";
    EnvironmentVariableName["MAX_QUERY_EXECUTION_TIME_ALT"] = "TYPEORM_MAX_QUERY_EXECUTION_TIME";
    EnvironmentVariableName["DEBUG"] = "DB_DEBUG";
    EnvironmentVariableName["DEBUG_ALT"] = "TYPEORM_DEBUG";
    EnvironmentVariableName["UUID_EXTENSION"] = "DB_UUID_EXTENSION";
    EnvironmentVariableName["UUID_EXTENSION_ALT"] = "TYPEORM_UUID_EXTENSION";
    EnvironmentVariableName["CACHE"] = "DB_CACHE";
    EnvironmentVariableName["CACHE_ALT"] = "TYPEORM_CACHE";
    EnvironmentVariableName["CACHE_ALWAYS_ENABLED"] = "DB_CACHE_ALWAYS_ENABLED";
    EnvironmentVariableName["CACHE_ALWAYS_ENABLED_ALT"] = "TYPEORM_CACHE_ALWAYS_ENABLED";
    EnvironmentVariableName["CACHE_OPTIONS"] = "DB_CACHE_OPTIONS";
    EnvironmentVariableName["CACHE_OPTIONS_ALT"] = "TYPEORM_CACHE_OPTIONS";
    EnvironmentVariableName["CACHE_DURATION"] = "DB_CACHE_DURATION";
    EnvironmentVariableName["CACHE_DURATION_ALT"] = "TYPEORM_CACHE_DURATION";
})(EnvironmentVariableName = exports.EnvironmentVariableName || (exports.EnvironmentVariableName = {}));
